zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/deserializer/BasicLongFilterPropertyDeserializer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.deserializer;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.NumericNode;
+import io.camunda.zeebe.gateway.protocol.rest.AdvancedLongFilter;
+import io.camunda.zeebe.gateway.protocol.rest.BasicAdvancedLongFilter;
+import io.camunda.zeebe.gateway.protocol.rest.BasicLongFilterProperty;
+import java.io.IOException;
+
+public class BasicLongFilterPropertyDeserializer
+    extends FilterDeserializer<BasicLongFilterProperty> {
+
+  public BasicLongFilterPropertyDeserializer(final ObjectMapper objectMapper) {
+    super(objectMapper);
+  }
+
+  @Override
+  public BasicAdvancedLongFilter deserialize(
+      final JsonParser parser, final DeserializationContext context) throws IOException {
+
+    final var treeNode = parser.getCodec().readTree(parser);
+    final var filter = new BasicAdvancedLongFilter();
+
+    if (treeNode instanceof NumericNode) {
+      filter.set$Eq(((NumericNode) treeNode).longValue());
```

Comment:
I'll try that, it should be possible.

Reply:
This works. So now we can automatically deserialize the implicit cases, too. It's pretty nice.
Interesting, that it can parse String to Long and Boolean automatically. This can be turned off with `ALLOW_COERCION_OF_SCALARS`, but I think it's okay.

Reply:
Nice 👍 Yeah, that is exactly why I proposed this 😄 There is so much magic going on in the Object Mapper and you can configure so many things that we cannot rebuild this easily. Relying on the mapper should make our lives a lot easier 👍